### PR TITLE
ContinuationGroup.asBoolean did not correctly implement Groovy truthiness

### PR DESCRIPTION
**Pull request by jglick**
Note that the test does not fail without the main source patch, for reasons unknown, though it does print the stack trace

```
java.lang.NullPointerException
	at com.cloudbees.groovy.cps.impl.ContinuationGroup.asBoolean(ContinuationGroup.java:50)
	at com.cloudbees.groovy.cps.impl.ElvisBlock$ContinuationImpl.jump(ElvisBlock.java:36)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
	at com.cloudbees.groovy.cps.impl.LocalVariableBlock$LocalVariable.get(LocalVariableBlock.java:33)
	at com.cloudbees.groovy.cps.LValueBlock$GetAdapter.receive(LValueBlock.java:30)
	at com.cloudbees.groovy.cps.impl.LocalVariableBlock.evalLValue(LocalVariableBlock.java:22)
	at com.cloudbees.groovy.cps.LValueBlock$BlockImpl.eval(LValueBlock.java:55)
	at com.cloudbees.groovy.cps.LValueBlock.eval(LValueBlock.java:16)
	at com.cloudbees.groovy.cps.Next.step(Next.java:58)
	at com.cloudbees.groovy.cps.Next.run(Next.java:49)
	at com.cloudbees.groovy.cps.Next$run.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:42)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:112)
	at com.cloudbees.groovy.cps.AbstractGroovyCpsTest.evalCPSonly(AbstractGroovyCpsTest.groovy:55)
	at com.cloudbees.groovy.cps.AbstractGroovyCpsTest.evalCPS(AbstractGroovyCpsTest.groovy:49)
	at com.cloudbees.groovy.cps.AbstractGroovyCpsTest$evalCPS.callCurrent(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:46)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:133)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:141)
	at com.cloudbees.groovy.cps.CpsTransformerTest.ternaryOp3(CpsTransformerTest.groovy:383)
```

@reviewbybees esp. @kohsuke

### CODE REVIEW

File: src/main/java/com/cloudbees/groovy/cps/impl/ContinuationGroup.java [hunk 1]
```diff
@@ -41,19 +40,6 @@ public Next then(Block exp, Env e, Continuation k) {
         return new Next(exp,e,k);
     }
 
-    /**
-     * Casts the value to boolean by following the Groovy semantics.
-     */
-    protected final boolean asBoolean(Object o) {
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/groovy/cps/impl/ContinuationGroup.java]: Should be safe for the private code

**Comment by oleg-nenashev** [comment]
:bee:

**Comment by tfennelly** [comment]
:bee:

**Comment by jglick** [comment]
@reviewbybees done

**Pull request merged**